### PR TITLE
Feature/developer applications

### PR DESCRIPTION
**Pull request by fisher60**
Useful snippet for testing:
```py
@router.get(
    "/test_user",
    summary="Login or create a test user account and retrieve a JWT for the user.",
    response_description="The test user's JWT",
    response_model=dict
)
async def create_or_login_test_user(user_id: str | None = None) -> dict:
    if user_id:
        user = await User.prisma().find_unique({
            "id": user_id
        })
    else:
        user = await User.prisma().create({
            "username": "internal_test_account"
        })

    return {
        "user_id": user.id,
        "token": generate_long_lived_jwt(user.id)
    }
  ```

### CODE REVIEW

**Comment by fisher60** [comment]
Developer applications core functionality is complete. We will still need to create a login method for persistent sessions on abandon auth. This is a good chance to start a UI for managing abandon auth accounts (and to assist logging in with OAuth).

**Comment by fisher60** [comment]
closes #11

File: abandonauth/routers/developer_application.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    """
+    dev_app = await DeveloperApplication.prisma().find_unique(
+        {
+            "id": login_data.id
+        }
+    )
+
+    if not dev_app or not verify_refresh_token(login_data.refresh_token, dev_app.refresh_token):
+        raise HTTPException(
+            status_code=HTTP_403_FORBIDDEN,
```
Comment (GDWR) [inline abandonauth/routers/developer_application.py:65]: 401?

**Review by GDWR — CHANGES_REQUESTED** [review]

File: abandonauth/dependencies/auth/refresh_token.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+def verify_refresh_token(plain_token, hashed_token):
+    return pwd_context.verify(plain_token, hashed_token)
+
+
+def get_refresh_token_hash(token):
+    return pwd_context.hash(token)
+
+
+def generate_refresh_token():
+    return pwd.genword(entropy=122, charset="ascii_72")
```
[the remark this answers is not in the document]
Comment (GDWR) [reply abandonauth/dependencies/auth/refresh_token.py:16]: although pyright didn't complain?

[the remark this answers is not in the document]
Comment (GDWR) [reply abandonauth/dependencies/auth/refresh_token.py:16]: ah, PyRight is happily inferring the types. I would suggest us to add 

```
[tool.pyright]
typeCheckingMode = "strict"
```

but there are many other issues that would not be fair for this PR.

**Pull request merged**